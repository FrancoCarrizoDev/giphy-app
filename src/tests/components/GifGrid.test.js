import { shallow } from "enzyme"
import { GifGrid } from "../../components/GifGrid"
import '@testing-library/jest-dom'
import React from 'react';
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs")
describe('Test in GifGrid Component', () => {

    test('should show the component in snapshot', () => {

        useFetchGifs.mockReturnValue({
            data:[],
            loading: true
        })
        const wrapper = shallow(<GifGrid category={'Pacman'}/>)
        expect(wrapper).toMatchSnapshot()
    })

    test('should show items with loading images useFetchGifs', () => {

        const gifs = [{
            id: 'ABC',
            url: 'https://google.com/image',
            title: 'ZXC'
        },
        {
            id: '1234',
            url: 'https://google.com/image',
            title: 'ZXC'
        }]

        useFetchGifs.mockReturnValue({
            data:gifs,
            loading: false
        })

        const wrapper = shallow(<GifGrid category={'Pacman'}/>)
        
        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('p').exists()).toBe(false)
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length)
        
    })
    
})
