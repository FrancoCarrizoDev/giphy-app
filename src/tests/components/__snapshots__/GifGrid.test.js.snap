// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in GifGrid Component should show items with loading images useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    Pacman
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="ZXC"
      url="https://google.com/image"
    />
    <GifGridItem
      id="1234"
      key="1234"
      title="ZXC"
      url="https://google.com/image"
    />
  </div>
</Fragment>
`;

exports[`Test in GifGrid Component should show the component in snapshot 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    Pacman
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
